classes/oai/ojs/OAIDAO.inc.php

```diff
@@ -252,72 +242,64 @@ public function _getRecordsRecordSet($setIds, $from, $until, $set, $submissionId
-				JOIN journals j ON (j.journal_id = a.context_id)
-			WHERE	i.published = 1 AND j.enabled = 1 AND a.status = ?
-				' . ($excludeJournals ? ' AND j.journal_id NOT IN (' . implode(',', $excludeJournals) . ')' : '') . '
-				' . (isset($journalId) ? ' AND j.journal_id = ?' : '') . '
-				' . (isset($sectionId) ? ' AND p.section_id = ?' : '') . '
-				' . ($from ? ' AND GREATEST(a.last_modified, i.last_modified) >= ' . $this->datetimeToDB($from) : '') . '
-				' . ($until ? ' AND GREATEST(a.last_modified, i.last_modified) <= ' . $this->datetimeToDB($until) : '') . '
-				' . ($submissionId ? ' AND a.submission_id = ?' : '') . '
-			UNION
-			SELECT	dot.date_deleted AS last_modified,
-				dot.data_object_id AS submission_id,
-				' . (isset($journalId) ? 'tsoj.assoc_id' : 'NULL') . ' AS assoc_id,' . '
-				' . (isset($sectionId) ? 'tsos.assoc_id' : 'NULL') . ' AS section_id,
-				NULL AS issue_id,
-				dot.tombstone_id,
-				dot.set_spec,
-				dot.oai_identifier
-			FROM	data_object_tombstones dot' . '
-				' . (isset($journalId) ? 'JOIN data_object_tombstone_oai_set_objects tsoj ON (tsoj.tombstone_id = dot.tombstone_id AND tsoj.assoc_type = ' . ASSOC_TYPE_JOURNAL . ' AND tsoj.assoc_id = ?)' : '') . '
-				' . (isset($sectionId) ? 'JOIN data_object_tombstone_oai_set_objects tsos ON (tsos.tombstone_id = dot.tombstone_id AND tsos.assoc_type = ' . ASSOC_TYPE_SECTION . ' AND tsos.assoc_id = ?)' : '') . '
-			WHERE	1=1
-				' . (isset($set) ? ' AND (dot.set_spec = ? OR dot.set_spec LIKE ?)' : '') . '
-				' . ($from ? ' AND dot.date_deleted >= ' . $this->datetimeToDB($from) : '') . '
-				' . ($until ? ' AND dot.date_deleted <= ' . $this->datetimeToDB($until) : '') . '
-				' . ($submissionId ? ' AND dot.data_object_id = ?' : '') . '
-			ORDER BY ' . $orderBy,
-            $params
-        );
+        $submissionsQuery = DB::table('submissions AS a')
+            ->select(DB::raw('GREATEST(a.last_modified, i.last_modified) AS last_modified,
+                                a.submission_id AS submission_id,
+                                j.journal_id AS journal_id,
+                                s.section_id AS section_id,
+                                i.issue_id,
+                                NULL AS tombstone_id,
+                                NULL AS set_spec,
+                                NULL AS oai_identifier'))
+            ->join('publications AS p', 'a.current_publication_id', '=', 'p.publication_id')
+            ->join('publication_settings AS psissue', function($join) {
+                $join->on('psissue.publication_id', '=', 'p.publication_id');
+                $join->where('psissue.setting_name', '=', DB::raw('\'issueId\''));
+                $join->where('psissue.locale', '=', DB::raw('\'\''));
+            })
+            ->join('issues AS i', function($join) {
+                $join->on(DB::raw('CAST(i.issue_id AS CHAR(20))'), '=', 'psissue.setting_value');
+            })
```

Comment:
Could this be used without `function($join)...`?
If so the result is the same, so you can decide and leave it as it is... :-)